plugins/entity-feedback-backend/src/service/router.ts

```diff
@@ -151,12 +151,23 @@ export async function createRouter(
   });
 
   router.get('/ratings/:entityRef', async (req, res) => {
+    const user = await identity.getIdentity({ request: req });
     const ratings = await dbHandler.getRatings(req.params.entityRef);
 
     const token = getBearerTokenFromAuthorizationHeader(
       req.header('authorization'),
     );
 
+    const entity = (
+      await catalogClient.getEntitiesByRefs(
+        {
+          entityRefs: [req.params.entityRef],
+          fields: ['spec.ownerRef'],
+        },
+        { token },
+      )
+    ).items[0];
```

Comment:
can this be empty? should we check that there is an item in the array?

Reply:
I debated this a lot. The CatalogApi promises that the output will be the same size and in the same order as the entityrefs parameter. Entries that are not found will be returned as null. So the array will never be empty but it's possible that items[0] is null in theory. BUT its loading feedback for a specific entity and if you go to an entity that doesn't exist it doesn't load anything and wouldn't even try loading the feedback so the entity will always exist.

Reply:
spec is technically optional so there is a chance spec could be null?

Reply:
yeah you already handle the spec being null in the query. I think your explanation makes total sense, I didn't know about the catalog API guarantee 😅 

I say leave it like this